src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertificateAssetDownloader.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Internal.Cryptography.Pal
+{
+    internal static class CertificateAssetDownloader
+    {
+        private static unsafe Interop.libcurl.curl_unsafe_write_callback s_writeCallback = CurlWriteCallback;
+
+        static CertificateAssetDownloader()
+        {
+            Interop.libcurl.curl_global_init(Interop.libcurl.CurlGlobalFlags.CURL_GLOBAL_ALL);
```

Comment:
Here is what @bartonjs, @ellismg, and I just discussed:

* System.Net.Http.Native.so [NEW]
  * exports thread-safe, idempotent EnsureLibCurlIsInitialized() [NEW]
  * requires caller to have already initialized OpenSsl (see below)

* System.Security.Cryptography.so  [EXISTING]
  * exports thread-safe, idempotent EnsureOpenSslIsInitialized [EXISTING]

Interop.Crypto.cctor -> calls EnsureOpenSslIsInitialized() [EXISTING]
Interop.Http.cctor -> calls EnsureOpenSslIsInitialized(); EnsureCurlIsIntialized() [NEW]

(And Interop.libcrypto and Interop.libcurl would do the same as Interop.Crypto and Interop.Http respective pending their elimination once everything is shimmed.)

@stephentoub does that make sense to you? I can work on this next to seed the curl shim and address this at the same time. 

One thing that got me worried just now is that it's unclear to me if the libcurl we use is guaranteed to be using the same libcrypto as us or if there could end up being two SxS libcryptos to initialize...


Interop.Crytp